qa/evil-tests/src/test/java/org/opensearch/action/main/EvilSystemPropertyTests.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.main;
+
+import java.io.IOException;
+
+import org.opensearch.Build;
+import org.opensearch.LegacyESVersion;
+import org.opensearch.Version;
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.common.Strings;
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class EvilSystemPropertyTests extends OpenSearchTestCase {
+
+    @SuppressForbidden(reason = "manipulates system properties for testing")
+    public void testToXContent_responseVersionOverride() throws IOException {
+        System.setProperty("opensearch.http.override_main_response_version", "true");
+        MainResponse response = new MainResponse("nodeName", Version.CURRENT,
+            new ClusterName("clusterName"), randomAlphaOfLengthBetween(10, 20), Build.CURRENT);
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        response.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        assertTrue(Strings.toString(builder).contains("\"number\":\"" + LegacyESVersion.V_7_10_2.toString() + "\","));
+        System.clearProperty("opensearch.http.override_main_response_version");
```

Comment:
Needs to be in a try/finally block or a test failure will pollute following tests.

Reply:
you can also do the clean up in @After

Reply:
Except that runs after all the tests, so if we add more tests it will break the subsequent ones in the same class.

I don't envy whomever has the username `@after` on GitHub :)

Reply:
updated with try/finally

Reply:
btw "After" runs after every test. "AfterClass" is run after all tests.

Reply:
You're right of course. I personally dislike doing a certain cleanup in `after` unless it was setup in `before`. Just to keep things symmetrical. In Ruby we have things like `context` in which you can do "when the value of this setting is X", setup in before, then cleanup after. Not sure it's possible here.